contracts/commands/BasicSellCommand.sol

```diff
@@ -0,0 +1,133 @@
+        uint256 cdpId;
+        uint16 triggerType;
+        uint256 execCollRatio;
+        uint256 targetCollRatio;
+        uint256 minSellPrice;
+        bool continuous;
+        uint64 deviation;
+    }
+
+    constructor(ServiceRegistry _serviceRegistry) BaseMPACommand(_serviceRegistry) {}
+
+    function decode(bytes memory triggerData) public pure returns (BasicSellTriggerData memory) {
+        return abi.decode(triggerData, (BasicSellTriggerData));
+    }
+
+    function isTriggerDataValid(uint256 _cdpId, bytes memory triggerData)
+        external
+        pure
+        returns (bool)
+    {
+        BasicSellTriggerData memory decodedTrigger = decode(triggerData);
+
+        (uint256 lowerTarget, ) = decodedTrigger.targetCollRatio.bounds(decodedTrigger.deviation);
+        return (_cdpId == decodedTrigger.cdpId &&
+            decodedTrigger.triggerType == 4 &&
+            decodedTrigger.execCollRatio < lowerTarget);
+    }
+
+    function isExecutionLegal(uint256 cdpId, bytes memory triggerData)
+        external
+        view
+        returns (bool)
+    {
+        BasicSellTriggerData memory decodedTriggerData = decode(triggerData);
+
+        (, uint256 nextCollRatio, uint256 nextPrice, ) = getVaultAndMarketInfo(cdpId);
+
+        return (decodedTriggerData.execCollRatio.wad() > nextCollRatio &&
+            decodedTriggerData.minSellPrice < nextPrice);
+    }
+
+    function execute(
+        bytes calldata executionData,
+        uint256 cdpId,
+        bytes memory triggerData
+    ) external {
+        BasicSellTriggerData memory decodedTriggerData = decode(triggerData);
+
+        validateTriggerType(decodedTriggerData.triggerType, 4);
+        validateSelector(MPALike.decreaseMultiple.selector, executionData);
+
+        executeMPAMethod(executionData);
+
+        if (decodedTriggerData.continuous) {
+            recreateTrigger(cdpId, decodedTriggerData.triggerType, triggerData);
+        }
+    }
+
+    function getDustLimit(bytes32 ilk) internal view returns (uint256 dustLimit) {
+        VatLike vat = VatLike(serviceRegistry.getRegisteredService(MCD_VAT_KEY));
+        (, , , , uint256 radDust) = vat.ilks(ilk);
+        uint256 wadDust = (radDust * WAD) / RAD;
+        return wadDust;
+    }
+
+    function getVaultInfo(uint256 cdpId) internal view returns (uint256 dustLimit) {
+        McdView mcdView = McdView(serviceRegistry.getRegisteredService(MCD_VIEW_KEY));
+        (, uint256 debt) = mcdView.getVaultInfo(cdpId);
+        return debt;
+    }
+
+    function isExecutionCorrect(uint256 cdpId, bytes memory triggerData)
+        external
+        view
+        returns (bool)
+    {
+        BasicSellTriggerData memory decodedTriggerData = decode(triggerData);
+        (, uint256 nextCollRatio, uint256 nextPrice, bytes32 ilk) = getVaultAndMarketInfo(cdpId);
+
+        uint256 dust = getDustLimit(ilk);
+
+        (, uint256 maxUnconditionallyAcceptedDebt) = dust.bounds(decodedTriggerData.deviation);
```

Comment:
the comment regarding contract-level parameter is left unresolved

Reply:
what comment ?

Reply:
the dust limit should not be bounded by `deviation` as it has no relation. create a contract level parameter that is the same for all users

Reply:
it is actually, If executor knows deviation it adjust planned collRatio in a way that 

`vaultDebtAfter.bounds(decodedTriggerData.deviation)` is exacly at `dust` (assuming that target would be below `dust`). 

but truth be told deviation should not be to `dust` but to `vaultDebtAfter`  


Generally I believe we do not need to artifically complicate it here.

Reply:
well, i don't think we should complicate it either, but my point stands. we've already had this discussion regarding liquidation ratio for basic buy. initial usage of `deviation` is for the bounds of the target ratio. using it for both target ratio and dust is confusing and doesn't make any sense since these are completely different values

Reply:
I disagree those are compeltely different values and that there is no relation. Situation with dust limit and basicSell is not the same as with liquidation ration and basicBuy

Reply:
i agree that it's not the same, but my point stands, you are using the same parameter for, essentially, two different calculations